sources/web/datalab/polymer/test/resizable-divider-test.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+declare function assert(condition: boolean, message: string): null;
+declare function fixture(element: string): any;
+
+/// <reference path="../node_modules/@types/mocha/index.d.ts" />
+/// <reference path="../node_modules/@types/chai/index.d.ts" />
+/// <reference path="../components/resizable-divider/resizable-divider.ts" />
+
+/*
+ * For all Polymer component testing, be sure to call Polymer's flush() after
+ * any code that will cause shadow dom redistribution, such as observed array
+ * mutation, wich is used by the dom-repeater in this case.
+ */
+
+function dragAndDrop(element: HTMLElement, x1: number, y1: number, x2: number, y2: number) {
+  const clickInfo: MouseEventInit = {
+    clientX: x1,
+    clientY: y1,
+  };
+  element.dispatchEvent(new MouseEvent('mousedown', clickInfo));
+
+  const moveInfo: MouseEventInit = {
+    clientX: x2,
+    clientY: y2,
+  };
+  document.dispatchEvent(new MouseEvent('mousemove', moveInfo));
+
+  document.dispatchEvent(new MouseEvent('mouseup'));
+}
+
+describe('<resizable-divider>', () => {
+  let testFixture: ResizableDividerElement;
+  let divider: HTMLDivElement;
+  let left: HTMLDivElement;
+  let right: HTMLDivElement;
+
+  beforeEach(() => {
+    testFixture = fixture('resizable-divider-fixture');
+    divider = testFixture.$.divider;
+    // Have to use querySelector to get the pane's slotted elements because
+    // they're in the inserted element's light DOM
+    left = testFixture.querySelector('#left-pane') as HTMLDivElement;
+    right = testFixture.querySelector('#right-pane') as HTMLDivElement;
+  });
+
+  it('inserts the given elements into panes', () => {
+
+    assert(!!left, 'left pane should exist under the divider element');
+    assert(!!right, 'right pane should exist under the divider element');
+
+    assert(left.innerText === 'left pane', 'left pane\'s contents should appear in first slot');
```

Comment:
The slots are neither defined nor filled in any particular order, they are identified by their names. Calling it the "first slot", as in this test's error message, is misleading. You could change the order of the definition or use to put the right slot first, and the code would still work, but then the error message would be wrong.